File: python/src/json_schema_dsl.py
JsonAny = {'type': 'Any'}
JsonString = {'type': 'String'}
JsonInteger = {'type': 'Integer'}
JsonNumber = {'type': 'Number'}
JsonBoolean = {'type': 'Boolean'}
JsonStringMap = {'type': 'StringMap'}


def JsonField(name, field_type, *filters):
    return {'name': name, 'field_type': field_type, 'filters': filters}


def JsonObject(*fields):
    return {'type': 'Object', 'fields': fields}


def JsonArray(element_type, *filters):
    return {'type': 'Array', 'element_type': element_type, 'filters': filters}

File: js/jsonSchemaDSL.js
const basicType = require('./basicType');
const jsonField = require('./jsonField');
const jsonArray = require('./jsonArray');
const jsonObject = require('./jsonObject');

JsonAny = basicType.anyType;
JsonString = basicType.stringType;
JsonInteger = basicType.integerType;
JsonBoolean = basicType.booleanType;
JsonStringMap = basicType.stringMap;
JsonFieldIn = function (name, fieldType, ...filters){
    return new jsonField.IncomingField(name, fieldType, filters);
};

JsonFieldOut = function (name, fieldType, ...filters){
    return new jsonField.OutgoingField(name, fieldType, filters);
};

JsonObjectIn = jsonObject.incomingObject;
JsonObjectOut = jsonObject.outgoingObject;
JsonArray = jsonArray.arrayClass;
